mono/mini/exceptions-amd64.c

```diff
@@ -920,28 +921,44 @@ mono_arch_handle_altstack_exception (void *sigctx, MONO_SIG_HANDLER_INFO_TYPE *s
 
 #ifndef DISABLE_JIT
 GSList*
-mono_amd64_get_exception_trampolines (gboolean aot)
+mono_amd64_get_exception_trampolines (gboolean aot, gboolean reg)
```

Comment:
The changes to this function are a little messy but do server I recall to consolidate some logic and micro optimize.

Reply:
A form of this is in another slightly messy PR (https://github.com/mono/mono/pull/14151) and if that goes first, these changes will be less messy.

I could also probably make the diff smaller by passing the info pointer/enum through the trampinfo.